WordPress/Classes/ViewRelated/Blog/SiteManagement/SettingsTitleSubtitleController.swift

```diff
@@ -0,0 +1,283 @@
+import UIKit
+import Gridicons
+
+
+/// Types the closures than can be provided as completion blocks
+typealias SettingsTitleSubtitleAction = ((SettingsTitleSubtitleController.Data) -> Void)
+
+
+/**
+ Presents a view controller with a textfiled and a textview, that can be used to create / edit a title and subtitle pair.
+*/
+final class SettingsTitleSubtitleController: UITableViewController {
+
+    /// The data to be presented on screen (i.e. title and subtitle).
+    final class Data {
+        var title: String?
+        var subtitle: String?
+
+        init(title: String?, subtitle: String?) {
+            self.title = title
+            self.subtitle = subtitle
+        }
+    }
+
+
+    /// String literals to be presented in an action confirmation alert
+    struct Confirmation {
+        let title: String
+        let subtitle: String
+        let actionTitle: String
+        let cancelTitle: String
+    }
+
+    fileprivate enum Sections: Int, CustomStringConvertible {
+        case name
+        case description
+
+        static let count: Int = {
+            var returnValue: Int = 0
+            while let _ = Sections(rawValue: returnValue) {
+                returnValue = returnValue + 1
+            }
+            return returnValue
+        }()
+
+        static func section(for index: Int) -> Sections {
+            guard index < count else {
+                return .name
+            }
+            return Sections(rawValue: index)!
+        }
+
+        var description: String {
+            switch self {
+            case .name:
+                return NSLocalizedString("Tag", comment: "Section header for tag name in Tag Details View.").uppercased()
```

Comment:
The controller is intended to be generic and reusable but the `Sections.description` is specific for `Tags`.  Maybe this was intended to be configurable?